frontend/src/components/AgentStatusBar.tsx

```diff
@@ -90,14 +91,19 @@ function AgentStatusBar() {
     }
   }, [curAgentState]);
 
+  let statusMessage = curStatusMessage.message;
+  if (!statusMessage || !statusMessage.length) {
+    statusMessage = AgentStatusMap[curAgentState].message;
+  }
+
```

Comment:
```suggestion
  const [statusMessage, setStatusMessage] = React.useState<string>(
    curStatusMessage.message,
  );

  React.useEffect(() => {
    if (!statusMessage || !statusMessage.length) {
      setStatusMessage(AgentStatusMap[curAgentState].message);
    }
  }, [curAgentState, statusMessage]);

```

Efficiently handle state changes within a React component.

Reply:
Thank you!